### PR TITLE
SPT-482:Update vocab schema to add extra validation

### PR DESCRIPTION
**Pull request by chaitanguttarlapalli**
What:

Differences have been identified between the SPOT MVP credential schema and the schema published in the [vocab repo]. Update the vocab site, to reflect the additions as detailed in the acceptance criteria.

Why:

The schema published in the vocab repo should reflect the SPOT checks, such that any CRI checking their work against schema will not be caught by any additional validation enforced by the SPOT schema.
We are updating the vocab schema so it becomes the source of truth and, ultimately, will be using the vocab schema in SPOT instead of the current MVP schema.

AcceptanceCriteria:

$ - Should have the following required claims sub, iss, nbf, and vc on a credential JWT

$.jti - Should be a non-empty string uri ( optional  but nonempty if present)

$.vc - Should have evidence  required  on identity check /SecurityCheck 

$.vc - Should have type required  on  all credentials

$.vc.evidence[].activityHistoryScore - Should have a max 4

$.vc.evidence[].ci - Should support either null or  array of  non empty strings

$.vc.evidence[].identityFraudScore -Should have a max max 3

$.vc.evidence[].strengthScore -Should have a max 4

$.vc.evidence[].validityScore -Should have a max 4

$.vc.evidence[].verificationScore - Should have a max 4

$.vc.credentialSubject[].birthDate[].value - Should be required and non empty

$.vc.credentialSubject[].name[].nameParts[].value - Should be required and nonempty

$.vc.credentialSubject[].name[].nameParts[].type - Should be required

### CODE REVIEW

File: v1/linkml-schemas/credentials.yaml [hunk 1]
```diff
@@ -93,10 +93,24 @@ classes:
     see_also:
       - ../json-schemas/IdentityCheckCredentialJWT.json
     attributes:
```
Comment (cmiddi) [inline v1/linkml-schemas/credentials.yaml]:
```suggestion
    slot_usage:
```

File: v1/linkml-schemas/credentials.yaml [hunk 2]
```diff
@@ -93,10 +93,24 @@ classes:
...
+      iss:
+        range: uri
+        description: The `iss` (issuer) claim identifies the principal that issued the JWT.
```
Comment (cmiddi) [inline v1/linkml-schemas/credentials.yaml]:
```suggestion
```

File: v1/linkml-schemas/credentials.yaml [hunk 3]
```diff
@@ -93,10 +93,24 @@ classes:
...
+        required: true
+      sub:
+        range: uri
+        description: The `sub` (subject) claim identifies the principal that is the subject of the JWT.
```
Comment (cmiddi) [inline v1/linkml-schemas/credentials.yaml]:
```suggestion
```

File: v1/linkml-schemas/credentials.yaml [hunk 4]
```diff
@@ -93,10 +93,24 @@ classes:
...
+        required: true
+      nbf:
+        range: integer
+        description: The `nbf` (not before) claim identifies the time the JWT is valid from.
```
Comment (cmiddi) [inline v1/linkml-schemas/credentials.yaml]:
```suggestion
```

File: v1/linkml-schemas/credentials.yaml [hunk 5]
```diff
@@ -93,10 +93,24 @@ classes:
...
+        required: true
       vc:
         range: IdentityCheckCredentialClass
         required: true
-
+      jti:
+        range: uri
```
Comment (cmiddi) [inline v1/linkml-schemas/credentials.yaml]:
```suggestion
```

**Review by cmiddi — COMMENTED** [review]
I think in this case using `slot_usage` may serve us better as there is duplication of descriptions and we are wanting to use the globally defined property just apply a more locked down pattern of usage.  Having reviewed this I am not sure we are entirely consistent elsewhere so I will look to address consistency in other model definitions.

File: v1/linkml-schemas/credentials.yaml (hunk 5 above)
Comment (chaitanguttarlapalli) [reply v1/linkml-schemas/credentials.yaml]: fixed

File: v1/linkml-schemas/credentials.yaml (hunk 4 above)
Comment (chaitanguttarlapalli) [reply v1/linkml-schemas/credentials.yaml]: fixed

File: v1/linkml-schemas/credentials.yaml (hunk 3 above)
Comment (chaitanguttarlapalli) [reply v1/linkml-schemas/credentials.yaml]: fixed

File: v1/linkml-schemas/credentials.yaml (hunk 2 above)
Comment (chaitanguttarlapalli) [reply v1/linkml-schemas/credentials.yaml]: fixed

File: v1/linkml-schemas/credentials.yaml (hunk 1 above)
Comment (chaitanguttarlapalli) [reply v1/linkml-schemas/credentials.yaml]: fixed

**Review by cmiddi — APPROVED** [review]
I believe these changes should now align with the acceptance criteria so happy to approve.

**Pull request merged**